### ISSUE TITLE
Explore the Azure Machine Learning workspace errors

### ISSUE THREAD

**Issue by 2good4hisowngood** [MicrosoftLearning/mslearn-azure-ml#54]
# Module: 01
#### Step: 02

Not sure if above is right, but it's [this url](https://microsoftlearning.github.io/mslearn-azure-ml/Instructions/02-Explore-Azure-Machine-Learning.html), creating the compute instance.

Description of issue:
When you create the compute instance, as outlined, and then run the job, it throws an error that the managed identity isn't set and you must fix that. If you go and set the managed identity to system assigned, and rerun the job, it fails with a new error.

Repro steps:

1. Provision DS11 compute instance
2. Run Job
3. Get error about managed identity > go to compute > set managed identity for the instance
4. Re-run job > Get failure with messaging as outlined below.

Job url (not sure if you can access it)
https://ml.azure.com/experiments/id/6bf99e55-62d4-4d70-b198-8ff721cbdb53/runs/3290d3dc-8583-4eaf-bcba-cbc7344edd44?wsid=/subscriptions/fe47f131-cb44-477f-82a5-7c19ca0fd15a/resourcegroups/cognative/providers/Microsoft.MachineLearningServices/workspaces/AzMl&tid=c6eca790-50bf-4c34-afe0-134b3944ca54#/?graphId=e959e2a8-d5f4-44e7-bf88-94c53ef0c11f&label=Regression+-+Automobile+Price+Prediction+%28Basic%29&path=%2Fexperiments%2Fid%2F6bf99e55-62d4-4d70-b198-8ff721cbdb53%2Fruns%2F3290d3dc-8583-4eaf-bcba-cbc7344edd44&runId=3290d3dc-8583-4eaf-bcba-cbc7344edd44

First job fail (managed identity error):
UserError: Identity of the specified managed compute fe47f131-cb44-477f-82a5-7c19ca0fd15a$cognative$azml$sean1 is not found. Set up managed identity, refer "https://docs.microsoft.com/en-us/azure/machine-learning/how-to-create-attach-compute-cluster?tabs=python#managed-identity".

Second Job fail (w/ system managed identity):
Select Columns in Dataset step output:

Execution failed. User process '/azureml-envs/azureml_8f317849db35f281450cf74333640b98/bin/python' exited with status code 1. Please check log file 'user_logs/std_log.txt' for error details. Error:   File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_host_executor.py", line 52, in execute_with_cli
    do_execute_with_env(parser, FolderRuntimeEnv())
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_host_executor.py", line 63, in do_execute_with_env
    ModuleReflector(parser.module_entry, env).exec(
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_reflector.py", line 397, in exec
    self._handle_exception(bex)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_reflector.py", line 471, in _handle_exception
    raise exception
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_reflector.py", line 381, in exec
    self._handle_output_ports(output_tuple, output_ports)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/core/logger.py", line 214, in wrapper
    ret = func(*args, **kwargs)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_reflector.py", line 438, in _handle_output_ports
    self._env.handle_output_port(data, annotation, output_value)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/env.py", line 192, in handle_output_port
    ErrorMapping.rethrow(
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/common/error.py", line 842, in rethrow
    raise err from e
azureml.studio.common.error.FailedToWriteOutputsError: Failed to write output files to disk. Reason: [Errno 5] Input/output error. Possibly caused by not enough space left on disk. Please upgrade VM Sku or use another compute with more disk space.

Warnings:
AzureMLCompute job failed
ExecutionFailed: [REDACTED]
	exit_codes: 1
	Appinsights Reachable: Some(true)

Execution failed. User process '/azureml-envs/azureml_8f317849db35f281450cf74333640b98/bin/python' exited with status code 1. Please check log file 'user_logs/std_log.txt' for error details. Error:   File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_host_executor.py", line 52, in execute_with_cli
    do_execute_with_env(parser, FolderRuntimeEnv())
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/s

Similar errors for Linear regression step:
Execution failed. User process '/azureml-envs/azureml_8f317849db35f281450cf74333640b98/bin/python' exited with status code 1. Please check log file 'user_logs/std_log.txt' for error details. Error:   File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/module_reflector.py", line 438, in _handle_output_ports
    self._env.handle_output_port(data, annotation, output_value)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/env.py", line 189, in handle_output_port
    self._do_handle_output_port(data, file_path, data_type)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/env.py", line 210, in _do_handle_output_port
    OutputHandler.handle_output(data, file_path, file_name, data_type,
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/handler/port_io_handler.py", line 162, in handle_output
    OutputHandler.handle_output_directory(data, file_path, file_name, data_type, saver)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/modulehost/handler/port_io_handler.py", line 127, in handle_output_directory
    saver(
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/core/io/model_directory.py", line 268, in save_model_to_directory
    ModelDirectory.create(
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/core/io/model_directory.py", line 131, in dump
    super().dump(save_to)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/azureml-envs/azureml_8f317849db35f281450cf74333640b98/lib/python3.8/site-packages/azureml/studio/core/io/any_directory.py", line 356, in os_io_error_handler
    raise UserError(f"Failed to write output files to disk. Reason: {str(OSError)}. "
azureml.studio.core.error.UserError: Failed to write output files to disk. Reason: <class 'OSError'>. Possibly caused by not enough space left on disk. Please upgrade VM Sku or use another compute with more disk space.

**Comment by Mihai-Ac** [comment]
Hello @2good4hisowngood 

I tried to reproduce the error(s) but I couldn´t , I tested with two different subscriptions on different tentants.

Could you try running through the exercise again with a fresh resource group and see if you are still seeing this error.

**Comment by 2good4hisowngood** [comment]
Hey Mihai, I wanted to test this thoroughly. I rebuilt the resources a few times, and even encoded it with [terraform](https://github.com/2good4hisowngood/terraform). But they had issues. Tf threw an error about Failed to get Workplace secrets, and to check my dns.

I destroyed the entire rg, and built it from scratch from the portal ui and was able to get it to run. I'm not sure why it occurred. I'll leave that repo visible for a few weeks, again, not sure why creating with ui worked this time when it didn't the first. ¯\_(ツ)_/¯

Anyway, have a great one!
